### PR TITLE
Document environment variables to disable instrumentation

### PR DESCRIPTION
**Pull request by marcotc**
This PR documents all `"DD_TRACE_#{integration}_ENABLED"` environment variables.

Today, these exists but are undocumented.

### CODE REVIEW

**Comment by marcotc** [comment]
It is much easier to review this PR using the "Display the rich diff" view:
Screenshot 2024-06-19 at 7 41 58 PM

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -1823,11 +1894,12 @@ end
...
-| `on_error` | `Proc` | Custom error handler invoked when a job raises an error. Provided `span` and `error` as arguments. Sets error on the span by default. Useful for ignoring transient errors. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |
+| Key        | Env Var | Type   | Description                                                                                                                                                                 | Default                                                           |
+| ---------- | - | ----- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------- |
+| `enabled`  | | `Bool` | Defines whether Sneakers should be traced. Useful for temporarily disabling tracing. `true` or `false`                                                                      | `true`                                                            |
```
Comment (TonyCTHsu) [inline docs/GettingStarted.md:1899]: Do we want to change the description to `Whether the integration should create spans.` ?

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -566,6 +595,12 @@ Datadog::Tracing.trace('outer') do
 end
 ```
 
+`options` are the following keyword arguments:
+
+| Key       | Env Var                         | Type   | Description                                  | Default |
+| --------- | ------------------------------- | ------ | -------------------------------------------- | ------- |
+| `enabled` | `DD_TRACE_CONCURRENT_RUBY_ENABLED` | `Bool` | Whether the integration should create spans. | `true` |
```
Comment (TonyCTHsu) [inline docs/GettingStarted.md:602]: The description is not accurate: `concurrent-ruby` integration does not generate spans.

**Review by TonyCTHsu — APPROVED** [review]
LGTM

Nitpick: Can we make the `enabled` as the first row of each table?

File: docs/GettingStarted.md (hunk 1 above)
Comment (marcotc) [reply docs/GettingStarted.md:1899]: Done! Also, this made me realize that there are few duplicate entries, which I removed.

File: docs/GettingStarted.md (hunk 2 above)
Comment (marcotc) [reply docs/GettingStarted.md]: Fixed, thank you!

**Review by brett0000FF — APPROVED** [review]
LGTM from Doc's side. 🚀

**Pull request merged**